jclklib/proxy/connect_ptp4l.cpp

```diff
@@ -268,43 +258,42 @@ void *ptp4l_event_loop( void *arg)
     event_subscription(NULL);
 
     while (1) {
-        if(is_ptp4l_running()) {
-            if (epoll_wait( epd, &epd_event, 1, 100) != -1) {
+        if (epoll_wait( epd, &epd_event, 1, 100) != -1) {
                 const auto cnt = sk->rcv(buf, bufSize);
+
+            if(cnt > 0) {
                 MNG_PARSE_ERROR_e err = msg.parse(buf, cnt);
                 if(err == MNG_PARSE_ERROR_OK)
                     event_handle();
-            }
-        }
-        else {
-            PrintError("Failed to connect to ptp4l. Retrying...");
-            SockUnix *sku = new SockUnix;
-            sk->close();
-            sku->close();
-            msg.clearData();
-
-            /* Reset TIME_STATUS_NP data when ptp4l is disconnected */
-            pe.master_offset = 0;
-            memset(pe.gm_identity, 0, sizeof(pe.gm_identity));
-            pe.synced_to_primary_clock = false;
-            pe.as_capable = 0;
-            notify_client();
-
-            while (1) {
-                if(is_ptp4l_running()) {
-                    PrintInfo("Connected to ptp4l via /var/run/ptp4l.");
-                    std::string uds_address;
-                    m_sk.reset(sku);
-                    uds_address = "/var/run/ptp4l";
-                    if(!sku->setDefSelfAddress() || !sku->init() ||
-                            !sku->setPeerAddress(uds_address))
-                        fprintf(stderr, "Fail to connect to ptp4l\n");
-
-                    sk = m_sk.get();
-                    event_subscription(NULL);
-                    break;
+            } else {
+                /* Reset TIME_STATUS_NP data when ptp4l is disconnected */
+                pe.master_offset = 0;
+                memset(pe.gm_identity, 0, sizeof(pe.gm_identity));
+                pe.synced_to_primary_clock = false;
+                pe.as_capable = 0;
+                notify_client();
+
+                while(1){
+                    if(msg_send(false))
+                        break;
+                    PrintError("Failed to connect to ptp4l. Retrying...");
+                    sleep(2);
```

Comment:
A question here, are we catching any Control-C for proxy ?

Reply:
yes, this is in another parallel thread. termination signal is controlled in main thread. 
Maybe we can do some experiment here: kill ptp4l, let the proxy goes into subscript looping, then ctrl+c to see whether proxy can be terminated or not.

Reply:
yes, when proxy is looping and retrying to connect to ptp4l, proxy is able to terminate with "ctrl + c".